### PR TITLE
[CA-1038] More tolerant of expired refresh tokens when unlinking

### PR DESCRIPTION
**Pull request by marctalbott**
Ticket: [CA-1038](https://broadworkbench.atlassian.net/browse/CA-1038)

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary. Documentation PRs only need 1 thumb.
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: bond_app/fence_api.py [hunk 1]
```diff
@@ -31,7 +32,8 @@ def delete_credentials_google(self, access_token, key_id):
         """
         headers = {'Authorization': 'Bearer ' + access_token}
         result = requests.delete(url=self.delete_service_account_url + key_id, headers=headers)
-        if result.status_code // 100 != 2:
+        logging.info("request: DELETE {} - status code: {}".format(self.delete_service_account_url, result.status_code))
+        if result.status_code // 100 != 2 or result.status_code // 100 != 4:
```
Comment (dvoet) [inline bond_app/fence_api.py]: this should be `and` not `or`

**Review by dvoet — CHANGES_REQUESTED** [review]

File: bond_app/fence_api.py [hunk 2]
```diff
@@ -31,7 +32,8 @@ def delete_credentials_google(self, access_token, key_id):
         """
         headers = {'Authorization': 'Bearer ' + access_token}
         result = requests.delete(url=self.delete_service_account_url + key_id, headers=headers)
-        if result.status_code // 100 != 2:
+        logging.info("request: DELETE {} - status code: {}".format(self.delete_service_account_url, result.status_code))
+        if result.status_code // 100 != 2 and result.status_code // 100 != 4:
```
Comment (gpolumbo-broad) [inline bond_app/fence_api.py:39]: Can we leave a comment here on why we are accepting `4xx` as OK.

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]

**Comment by gpolumbo-broad** [comment]
And I'm still an advocate of directly deleting the item from the cache when we know we can remove it at deletion time rather than waiting for the expiry logic to purge it, but I don't have a good technical or logical reason why need to do it, so it's fine.  To me, it's sort of like dropping a piece of trash on the street when I see the street sweeper coming.  It's going to get swept up, so I can just leave it, but I'm probably going to pick it up and throw it in the trash. :)

**Comment by marctalbott** [comment]
> And I'm still an advocate of directly deleting the item from the cache when we know we can remove it at deletion time rather than waiting for the expiry logic to purge it, but I don't have a good technical or logical reason why need to do it, so it's fine. To me, it's sort of like dropping a piece of trash on the street when I see the street sweeper coming. It's going to get swept up, so I can just leave it, but I'm probably going to pick it up and throw it in the trash. :)

We were directly deleting SA keys that had been associated with user ids from the cache. We don't do that anymore after #155 because of the discrepancy it was causing between pets and users calling Bond. That's why this PR removes that direct deletion, since there's no longer anything to delete

**Review by gpolumbo-broad — APPROVED** [review]

**Review by dvoet — APPROVED** [review]

**Pull request merged**